tmarktaylor/phmdoctest
# Merge 1.3.0 changes into master

Version 1.3.0

- Add main.testfile().
- Add testfile_creator and testfile_tester fixtures.
- Bugfix- Issue- Generated test name has output_NN when skip
  directive on output block.
- Bugfix- Issue- mark.skipif example code causes pytest AST fail at
  assertion rewrite time. Happens on skipped Python version.
  Replaced with code that compiles on the skipped version.
- Drop Python 3.7 add Python 3.10.

tests:

- Add mode=0o700 to mkdir() calls in test .yml files.
- Run tests in virtual envirionments. ci.yml.
- Add test_details.py.
- Add Appveyor to CI to show pytest items.
- Rework requirements files. Add tests.
- Refactor new fixtures to conftest.py.
- Rework/refactor quick_links test logic.
- Add test to find trailing spaces in sources.
- Tox no longer used in test suite.
- Add codecov.yml to set threshold.

docs:
- Bugfix- Issue- Markdown header level out of sequence.
- Loosen doc dependencies.
- Fenced code block info_string pycon -> py.
- Sphinx with myst_parser for docs.

style:
- Style/pep8/inspection fixes.
- Path and open changes.
- Remove trailing spaces from ~25 files.
